tests/compiletest.rs

```diff
@@ -88,6 +88,23 @@ fn miri_pass(sysroot: &Path, path: &str, target: &str, host: &str, need_fullmir:
         return;
     }
 
+    let target_lib_dir = sysroot.join("lib").join("rustlib").join(target).join("lib");
+    let has_std = std::fs::read_dir(target_lib_dir)
+        .expect("invalid target")
+        .map(|entry| entry.unwrap())
+        .filter(|entry| entry.file_type().unwrap().is_file())
+        .filter_map(|entry| entry.file_name().into_string().ok())
+        .any(|file_name| file_name.starts_with("libstd") && file_name.ends_with(".rlib"));
```

Comment:
Wow, I had no idea that there are targets that do not have a libstd...

Can you make this part of `for_all_targets` instead?